### PR TITLE
[SMALLFIX] Fixing concurrentListStatus test

### PR DESCRIPTION
**Pull request by yuzhu**
The reason the concurrentListStatus test was failing was because there is a period of time where the `inode` can be found through the tree traversal e.g. `getChildren`, but can not be found in the global list of inode list.

### CODE REVIEW

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by yuzhu** [comment]
jenkins, test this please

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 1]
```diff
@@ -685,6 +685,8 @@ public CreatePathResult createPath(RpcContext rpcContext, LockedInodePath inodeP
         }
         lastInode.setPinned(currentInodeDirectory.isPinned());
 
+        // Update state while holding the write lock.
+        mInodes.add(lastInode);
         if (!currentInodeDirectory.addChild(lastInode)) {
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: Does the inode have to be removed in this block?

Comment (yuzhu) [reply core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: done

**Pull request merged**